lazy_dataset/core.py

```diff
@@ -1956,10 +1972,12 @@ def __init__(self, slice, input_dataset: Dataset):
         self.input_dataset = input_dataset
 
     def copy(self, freeze=False):
-        return self.__class__(
-            input_dataset=self.input_dataset.copy(freeze=freeze),
-            slice=self._slice,
-        )
+        # Use __new__ to not deepcopy attributes
```

Comment:
`# Use __new__ to not deepcopy attributes` -> `# Use __new__ to avoid __init__ overhead`?